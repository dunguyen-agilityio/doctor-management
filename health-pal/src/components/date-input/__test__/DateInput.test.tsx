import { act, fireEvent, render } from '@utils-test'

import React, { createRef } from 'react'

import DateInput from '..'

describe('DateInput', () => {
  const setup = (props = {}) => {
    return render(<DateInput {...props} />)
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('match snapshot', () => {
    const date = new Date(2025, 0, 1)
    const { toJSON } = setup({ value: date })
    expect(toJSON()).toMatchSnapshot()
  })

  it('renders with formatted initial value', () => {
    const date = new Date(2025, 0, 1)
    const { getByLabelText } = setup({ value: date })
    expect(getByLabelText('Select date').props.defaultValue).toBe('01/01/2025')
  })

  it('shows date picker when clicked', async () => {
    const { getByTestId } = setup()
    act(() => {
      fireEvent.press(getByTestId('trigger-input'))
    })
    expect(getByTestId('calendar')).toBeTruthy()
  })

  it('displays error message when provided', () => {
    const { getByText } = setup({ errorMessage: 'Invalid date' })
    expect(getByText('Invalid date')).toBeTruthy()
  })

  it('input is non-editable and pointerEvents="none"', () => {
    const { getByLabelText } = setup()
    const input = getByLabelText('Select date')
    expect(input.props.editable).toBe(false)
    expect(input.props.pointerEvents).toBe('none')
  })

  it('supports imperative focus with ref', () => {
    const ref = createRef<any>()
    const { getByTestId } = render(<DateInput ref={ref} />)

    act(() => {
      ref.current?.focus()
    })

    expect(getByTestId('calendar')).toBeTruthy()
  })
})
