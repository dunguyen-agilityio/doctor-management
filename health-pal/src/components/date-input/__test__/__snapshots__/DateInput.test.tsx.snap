// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput match snapshot 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    accessibilityHint="Select a date from the calendar"
    aria-expanded={false}
    aria-label="Date picker"
    focusVisibleStyle={{}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="dialog"
    style={
      {
        "alignItems": "center",
        "backgroundColor": "hsla(0, 0%, 100%, 1)",
        "borderBottomColor": "transparent",
        "borderBottomWidth": 1,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 1,
        "borderRightColor": "transparent",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "transparent",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "height": 48,
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    <View
      style={[]}
    >
      <View
        style={[]}
      >
        <View
          style={[]}
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="19"
            bbWidth="18"
            fill="none"
            focusable={false}
            height="19"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 19,
                  "width": 18,
                },
              ]
            }
            testID="left-icon"
            vbHeight={19}
            vbWidth={18}
            width="18"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M6 2V4.25"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                    "strokeMiterlimit",
                  ]
                }
                stroke={
                  {
                    "payload": 4288455599,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeMiterlimit={10}
                strokeWidth="1.5"
              />
              <RNSVGPath
                d="M12 2V4.25"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                    "strokeMiterlimit",
                  ]
                }
                stroke={
                  {
                    "payload": 4288455599,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeMiterlimit={10}
                strokeWidth="1.5"
              />
              <RNSVGPath
                d="M2.625 7.3175H15.375"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                    "strokeMiterlimit",
                  ]
                }
                stroke={
                  {
                    "payload": 4288455599,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeMiterlimit={10}
                strokeWidth="1.5"
              />
              <RNSVGPath
                d="M15.75 6.875V13.25C15.75 15.5 14.625 17 12 17H6C3.375 17 2.25 15.5 2.25 13.25V6.875C2.25 4.625 3.375 3.125 6 3.125H12C14.625 3.125 15.75 4.625 15.75 6.875Z"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                    "strokeMiterlimit",
                  ]
                }
                stroke={
                  {
                    "payload": 4288455599,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeMiterlimit={10}
                strokeWidth="1.5"
              />
              <RNSVGPath
                d="M8.99686 10.775H9.00359"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4288455599,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              />
              <RNSVGPath
                d="M6.22049 10.775H6.22723"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4288455599,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              />
              <RNSVGPath
                d="M6.22049 13.025H6.22723"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4288455599,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <TextInput
          accessibilityHint="Opens a date picker to choose a date"
          aria-label="Select date"
          defaultValue="01/01/2025"
          editable={false}
          focusVisibleStyle={{}}
          focusable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="#9ca3af"
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#f9fafb",
              "borderBottomColor": "#d1d5db",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderBottomWidth": 1,
              "borderLeftColor": "#d1d5db",
              "borderLeftWidth": 1,
              "borderRightColor": "#d1d5db",
              "borderRightWidth": 1,
              "borderStyle": "solid",
              "borderTopColor": "#d1d5db",
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "borderTopWidth": 1,
              "color": "#374151",
              "flex": 1,
              "fontFamily": "Inter_400Regular",
              "fontSize": 14,
              "height": 45,
              "minWidth": 0,
              "paddingBottom": 12,
              "paddingLeft": 54,
              "paddingRight": 16,
              "paddingTop": 12,
              "position": "relative",
            }
          }
          testID="trigger-input"
          textContentType="dateTime"
        />
      </View>
    </View>
  </View>,
]
`;
